neptune_lightgbm/impl/__init__.py

```diff
@@ -79,7 +94,79 @@ def __init__(self, run: 'neptune.Run', base_namespace=''):
         self._run['source_code/integrations/neptune-lightgbm'] = __version__
 
     def __call__(self, env):
-        for name, loss_name, loss_value, _ in env.evaluation_result_list:
-            channel_name = '{}{}_{}'.format(self._base_namespace, name, loss_name)
+        # TODO:
+        # (data_name, eval_name, val, is_higher_better): `lightgbm.basic.Booster.__inner_eval`
+        # ('cv_agg', k, np.mean(v), metric_type[k], np.std(v)): `lightgbm.engine._agg_cv_result`
+        for name, loss_name, loss_value, *_ in env.evaluation_result_list:
```

Comment:
Here we're interested in only three first values of every entry of `evaluation_result_list` which is generated differently for different `NeptuneCallback` use cases:

#### for `train` and `LGBMClassifier`

It uses [`__inner_eval`](https://github.com/microsoft/LightGBM/blob/master/python-package/lightgbm/basic.py#L3382) and has 4 values of format: `(data_name, eval_name, val, is_higher_better)`.

#### for `cv`
It uses [`_agg_cv_result`](https://github.com/microsoft/LightGBM/blob/fcd2453514a6ecd780121e4ea31a227559145451/python-package/lightgbm/engine.py#L379) and has 5 values of format: `('cv_agg', k, np.mean(v), metric_type[k], np.std(v))`

Does this code makes sense?
Example results:
* https://app.stage.neptune.ai/wajcha/domain-2-project/e/DOM-977/all?path=foo%2Ffoo2&attribute=training_multi_logloss
* https://app.stage.neptune.ai/wajcha/domain-2-project/e/DOM-978/all?path=foo%2Flgbm_cv%2Ffoo2&attribute=cv_agg_multi_error
* https://app.stage.neptune.ai/wajcha/domain-2-project/e/DOM-979/all?path=foo%2Ffoo2&attribute=validation_multi_error
*

Reply:
* Let's refactor: `channel_name` -> `log_name`.
* For **train**, indeed, we need first three values. However, your structure is incorrect -> pls check here: https://github.com/neptune-ai/neptune-client/blob/kk/lgbm/integrations/lightgbm/neptune_lightgbm/impl/__init__.py#L117:L130
(`log_name = "{}{}/{}".format(self._base_namespace, dataset, metric)`)

* for **CV**, we need first three and last values, as we need to log mean and std. Can you adapt it to handle this as well? Pls consult draft implementation for the run structure: https://github.com/neptune-ai/neptune-client/blob/kk/lgbm/integrations/lightgbm/neptune_lightgbm/impl/__init__.py#L117:L130
(`log_val_name = "{}{}/{}/val".format(self._base_namespace, dataset, metric)`)